src/coreclr/jit/importercalls.cpp

```diff
@@ -10062,7 +10063,7 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
                                 assert(strcmp(className, "Vector`1") == 0);
                                 result = NI_Vector_GetCount;
                             }
-                            else if (gtIsRecursiveCall(method))
```

Comment:
Is this a general issue around all `gtIsRecursiveCall` that should maybe be fixed? The intent for that API was basically "am I calling myself" and calling some base or derived virtual method target shouldn't count as recursion.

Reply:
the current impl of `gtIsRecursiveCall` is "am I calling whatever is root method now" - it's fine for most cases, but in case of must-expand intrinsics, we only care about "method intentionally invoking itself"